src/test/groovy/graphql/schema/idl/SchemaPrinterTest.groovy

```diff
@@ -1342,9 +1342,195 @@ type Query {
 '''
     }
 
+    def "can print extend schema block when AST printing enabled"() {
+        def sdl = '''
+            directive @schemaDirective on SCHEMA
+            
+            """
+            My schema block description
+            """
+            schema {
+                mutation: MyMutation
+            }
+            
+            extend schema @schemaDirective {
+                query: MyQuery
+            }
```

Comment:
Note: although there is an implicit `schema` block if it is not specified, if you want to `extend schema` you MUST also have an explicit `schema` block defined in the schema.